src/core/utils/Lazy.ts

```diff
@@ -0,0 +1,41 @@
+export interface ILazyInitializer<T> {
+  (): T
+}
+
+export class Lazy<T> {
+  private instance: T | null = null
+  private initializer: ILazyInitializer<T>
+
+  constructor(initializer: ILazyInitializer<T>) {
+      this.initializer = initializer
+  }
+
+  public get value(): T {
```

Comment:
Maybe call this `get` for consistency?

Reply:
`value` is a getter, so you don't have to call it, so i don't think the noun value is better here. It is get below because you can't use `async` with `get` - does that make sense?